admin/src/helpers/internal-fetch.spec.js

```diff
@@ -0,0 +1,59 @@
+import internalFetch from './../helpers/internal-fetch';
+import session from './../helpers/Session';
+
+describe('internal fetch', () => {
+  it('Respond with the request body if the token is not expired', async () => {
+    session.getToken = jest.fn().mockImplementationOnce(() => 'token');
+    global.fetch = jest.fn().mockImplementationOnce((_, options) => {
+      return Promise.resolve({
+        ok: true,
+        status: 200,
+        json: async () => {
+          return { test: 'test' };
+        }
+      });
+    });
+
+    const res = await internalFetch('');
+    expect(res.ok).toEqual(true);
+    expect(res.status).toEqual(200);
+    expect(await res.json()).toEqual({ test: 'test' });
+  });
+
+  it('Should redirect and destroy the session if JWT is expired', async () => {
+    session.getToken = jest.fn().mockImplementationOnce(() => 'token');
```

Comment:
I'm mocking session using `jest.mock('../../helpers/Session');` in other places, but it's ok too :)